Rubberduck.Parsing/Symbols/IdentifierReference.cs

```diff
@@ -83,46 +74,26 @@ public bool IsIgnoringInspectionResultFor(string inspectionName)
                        && ((IgnoreAnnotation) annotation).IsIgnored(inspectionName));
         }
 
-        private readonly bool _hasExplicitLetStatement;
-        public bool HasExplicitLetStatement { get { return _hasExplicitLetStatement; } }
+        public bool HasExplicitLetStatement { get; }
 
         public bool HasExplicitCallStatement()
         {
             return Context.Parent is VBAParser.CallStmtContext && ((VBAParser.CallStmtContext)Context).CALL() != null;
         }
 
-        private bool? _hasTypeHint;
         public bool HasTypeHint()
-        {
-            if (_hasTypeHint.HasValue)
-            {
-                return _hasTypeHint.Value;
-            }
-
-            string token;
-            return HasTypeHint(out token);
```

Comment:
Doesn't removing this eliminate the caching?

Reply:
If we want to take advantage of caching, the correct solution here is to introduce a `Lazy` instead

Reply:
`Lazy` as in [Lazy<T> Class](https://msdn.microsoft.com/en-us/library/dd642331(v=vs.110).aspx)?

Reply:
@IvenBach the very one